syntax = "proto3";

package glog;

option go_package="glog/pb";

enum LogLevel {
  INFO = 0;
  WARNING = 1;
  ERROR = 2;
  DEBUG = 4;
}

message LogMessage {
  LogLevel level = 1;
  string service_name = 2;
  string message = 3;
  int64 timestamp = 4;
  string trace_id = 5;
}

message LogResponse {
  bool success = 1;
  string message = 2;
}

service LoggingService {
  rpc StreamLogs(stream LogMessage) returns (LogResponse);
}
